src/genie/libs/parser/asa/show_version.py

```diff
@@ -0,0 +1,730 @@
+# Parser for 'show version'
+# =============================================
+class ShowVersion(ShowVersionSchema):
+    """Parser for
+        * show version
+    """
+
+    cli_command = 'show version'
+
+    def cli(self, output=None):
+        if output is None:
+            # excute command to get output
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+
+        version_dict = {}
+
+        # Cisco Adaptive Security Appliance Software Version 9.8(4)10
+        p0 = re.compile(r'^.+ Software Version (?P<asa_version>[\S]+)')
+
+        # Firepower Extensible Operating System Version 2.2(2.121)
+        p1 = re.compile(r'^.+System Version (?P<firepower_version>[\S]+)')
+
+        # Device Manager Version 7.8(2)
+        p2 = re.compile(r'^.+Manager Version (?P<asdm_version>[\S]+)')
+
+        # Compiled on Tue 20-Aug-19 12:46 PDT by builders
+        # Compiled on Thu 20-Jan-12 04:05 by builders
+        p3 = re.compile(
+            r'^Compiled on (?P<compiled_date>\w{3} \S+ \d{1,2}:\d{1,2}.+) by (?P<compiled_by>\w+)')
+
+        # System image file is "boot:/asa984-10-smp-k8.bin"
+        p4 = re.compile(r'^System image.+"(?P<system_image>\S+)"')
+
+        # Config file at boot was "startup-config"
+        p5 = re.compile(r'^Config file.+"(?P<boot_config_file>\S+)"')
+
+        # ciscoasa up 1 day 12 hours
+        p6 = re.compile(r'^(?P<hostname>\S+) up (?P<uptime>.+)')
+
+        # Hardware:   ASAv, 2048 MB RAM, CPU Xeon E5 series 3491 MHz
+        p7 = re.compile(r'^Hardware:\s+(?P<platform>\S+),'
+                        r'\s+(?P<mem_size>[\w\s]+) RAM,'
+                        r'\s+CPU (?P<processor_type>[\w\s]+)')
+
+        # Model Id:   ASAv10
+        p8 = re.compile(r'^Model Id:\s*(?P<asa_model>\S+)')
+
+        # Internal ATA Compact Flash, 8192MB
+        # Slot 1: ATA Compact Flash, 8192MB
+        p9 = re.compile(r'^(?P<disk_name>\S+|Slot \d+):? '
+                        r'(?P<type_of_disk>[\w\s]+), '
+                        r'(?P<disk_size>\S+)')
+
+        # BIOS Flash Firmware Hub @ 0x0, 0KB
+        p10 = re.compile(r'^BIOS Flash (?P<bios_flash>.+)')
+
+        # 0: Ext: Management0/0       : address is 5001.0003.0000, irq 11
+        # 1: Ext: GigabitEthernet0/0  : address is 5001.0003.0001, irq 11
+        # 2: Ext: GigabitEthernet0/1  : address is 5001.0003.0002, irq 10
+        # 3: Ext: GigabitEthernet0/2  : address is 5001.0003.0003, irq 10
+        # 4: Ext: GigabitEthernet0/3  : address is 5001.0003.0004, irq 11
+        # 5: Ext: GigabitEthernet0/4  : address is 5001.0003.0005, irq 11
+        # 6: Ext: GigabitEthernet0/5  : address is 5001.0003.0006, irq 10
+        # 7: Ext: GigabitEthernet0/6  : address is 5001.0003.0007, irq 10
+        p11 = re.compile(r'^\s*(?P<intf_number>\d+): '
+                         r'Ext: (?P<interface>[\w\/\.\-]+)\s*: address is '
+                         r'(?P<mac_addr>[0-9A-Fa-f]{4}\.[0-9A-Fa-f]{4}\.[0-9A-Fa-f]{4}), '
```

Comment:
If you'd prefer, `(?P<mac_addr>[0-9A-Fa-f]{4}\.[0-9A-Fa-f]{4}\.[0-9A-Fa-f]{4})` can be shortened to something like `(?P<mac_addr>[\w\.]+)`. Not a required change